docs/operating-scylla/procedures/cluster-management/handling-membership-change-failures.rst

```diff
@@ -0,0 +1,202 @@
+Handling cluster membership change failures
+*******************************************
+
+A failure may happen in the middle of a cluster membership change (that is: bootstrap, decommission, removenode or replace), such as loss of power. If that happens, you should ensure that the cluster is brought back to a consistent state as soon as possible. Further membership changes might be impossible until you do so.
```

Comment:
First of all, let me say how much I appreciate this work. However, given all the comments we receive on this pull request, I think, we need to explain why they will be impossible: 
the failed node may remain a member of Raft quorum and prevent the cluster from committing changes to its Raft state. 
I would also mention the kind of behaviour to expect in this case: nodetool operation timeout. It is not the best behaviour and perhaps we should provide a user-friendly error message to avoid long and futile waits.

Reply:
I am also wondering what specific case we have in mind here, since in most cases the node joins as a non-voter, and only then becomes a voter.

Reply:
I think it is worth explaining that in reality a single failed node hardly makes a difference. But if operation failures accumulate, they may render the cluster useless.

Reply:
Non-voters don't prevent further topology changes.

I'm aware of at least one case where a failed decommission prevents further topology changes, and it is not related to Raft. If a node crashes during decommission after `set_bootstrap_state` in `leave_ring()` (I placed `abort` there to reproduce the problem):
```cpp
future<> storage_service::leave_ring() {
    co_await _sys_ks.local().set_bootstrap_state(db::system_keyspace::bootstrap_state::NEEDS_BOOTSTRAP);
    std::abort();
    co_await mutate_token_metadata([this] (mutable_token_metadata_ptr tmptr) {
        auto endpoint = get_broadcast_address();
        tmptr->remove_endpoint(endpoint);
        return update_pending_ranges(std::move(tmptr), format("leave_ring {}", endpoint));
    });

    auto expire_time = _gossiper.compute_expire_time().time_since_epoch().count();
    co_await _gossiper.add_local_application_state(gms::application_state::STATUS,
            versioned_value::left(co_await _sys_ks.local().get_local_tokens(), expire_time));
    auto delay = std::max(get_ring_delay(), gms::gossiper::INTERVAL);
    slogger.info("Announcing that I have left the ring for {}ms", delay.count());
    co_await sleep_abortable(delay, _abort_source);
}
```

The node will then refuse to restart, and further nodes will refuse to bootstrap because they need to contact all nodes to bootstrap e.g. to do the CDC generation write which uses CL=ALL, including the node which failed to decommission (it's still a NORMAL member in this case):
```
init - Startup failed: exceptions::unavailable_exception (Cannot achieve consistency level for cl ALL. Requires 3, alive 2)
```

There easily might be more such places in decommission where a failure may render a node unusable and the cluster in a weird state, I'm pretty sure we're not aware of them all.

I'm not sure we should write out detailed explanations of possible ways of how topology requests might be impossible after a failure because:
1. that's not the goal of the document, the goal is to explain how to handle failures / bring the cluster to a good state
2. we don't/cannot know all the possible ways. I wouldn't "expect" anything "nice" or predictable to happen until the failure is handled.